### PR TITLE
Restore `foreignKey` and `as` behaviour - reverts PR #5957

### PR DESCRIPTION
**Pull request by overlookmotel**
PR to revert PR #5957, as discussed here: https://github.com/sequelize/sequelize/issues/6023#issuecomment-225798272

### CODE REVIEW

File: test/integration/associations/belongs-to-many.test.js [hunk 1]
```diff
@@ -470,14 +470,7 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), function() {
...
 
-      var self = this;
-      return this.UserTask.sync({ force: true })
-      .then(function() {
-        return self.user.setActiveTasks(self.tasks);
-      })
-      .then(function() {
-        return expect(self.user.countActiveTasks({})).to.eventually.equal(1);
-      });
+      return expect(this.user.countActiveTasks({})).to.eventually.equal(1);
```
Comment (overlookmotel) [inline test/integration/associations/belongs-to-many.test.js]: @sushantdhiman Can you explain what this change did? Was it a fix of the test, unrelated to the `foreignKey`/`as` change?

File: test/integration/associations/belongs-to-many.test.js [hunk 2]
```diff
@@ -493,23 +486,18 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), function() {
...
-          }),
-          this.Task.create().then(function (task) {
-            return user.addStartedTask(task, {
-              started: true
-            });
-          })
-        );
-      })
-      .then(function () {
+      return Promise.join(
```
Comment (overlookmotel) [inline test/integration/associations/belongs-to-many.test.js]: @sushantdhiman Can you explain what this change did? Was it a fix of the test, unrelated to the `foreignKey`/`as` change?

File: test/integration/associations/belongs-to-many.test.js (hunk 1 above)
Comment (sushantdhiman) [inline test/integration/associations/belongs-to-many.test.js]: Yes, It was

File: test/integration/associations/belongs-to-many.test.js (hunk 2 above)
Comment (sushantdhiman) [inline test/integration/associations/belongs-to-many.test.js]: This one too

**Comment by overlookmotel** [comment]
Random Travis failure. Trying again...

File: test/integration/associations/belongs-to-many.test.js (hunk 1 above)
Comment (overlookmotel) [inline test/integration/associations/belongs-to-many.test.js]: OK, so it should be reverted then along with rest of the PR? (for now, anyway!)

**Comment by overlookmotel** [comment]
Think this is ready to merge now...

**Comment by overlookmotel** [comment]
Rebased on latest master. Things are moving pretty fast, so @mickhansen @janmeier if you're happy with this PR, would it be possible to merge it before it gets in conflict with master again?

**Comment by janmeier** [comment]
Thanks for pinging me @overlookmotel, I wanted to take my time reviewing to make sure we got it right this time, but then promptly forgot about it :)

**Comment by overlookmotel** [comment]
@janmeier No worries. Thanks for merging!

**Pull request merged**